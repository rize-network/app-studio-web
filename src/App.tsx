import React, { Suspense } from 'react';
import { Route, Routes, useLocation } from 'react-router-dom';
import { ComponentRouter } from 'src/routes/routes';
import { View, Text } from 'app-studio';

export const App: React.FC = () => {
  const location = useLocation();
  return (
    <Suspense
      fallback={
        <View
          height="100vh"
          display="flex"
          alignItems="center"
          justifyContent="center"
        >
          <Text>Loading...</Text>
        </View>
      }
    >
      <Routes location={location}>
        {ComponentRouter.map((item, index) => (
          <Route key={index} path={item.path} element={item.element} />
        ))}
      </Routes>
    </Suspense>
  );
};

export default App;
