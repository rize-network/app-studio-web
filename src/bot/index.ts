import { Bot } from './Bot';
import { DocuCode } from './DocuCode';

async function main() {
  const projectName = 'MDX Documentation';
  const descriptionPath = 'docs/description.md';

  // Get componentName from command line arguments
  const componentName = process.argv[2]; // Assuming it's the first argument passed
  if (!componentName) {
    console.error('Please provide a component name as an argument.');
    process.exit(1);
  }

  console.log(`Generating documentation for component: ${componentName}`);

  const basePath = 'src/components';
  const propsPath = `src/data/props/${componentName}.props.json`;
  const componentFolder = `${basePath}/${componentName}`;
  const componentPath = `${basePath}/${componentName}/${componentName}`;

  const assistantGPT = new Bot(componentName);

  // Implementation for Documentation of the code
  console.log('Initializing documentation process...');
  const docuCode = new DocuCode(componentName);
  const assistantDocumentation = await assistantGPT.init(
    projectName,
    [],
    descriptionPath,
    'docu'
  );

  await docuCode.processDirectory(componentFolder, assistantDocumentation.id);

  // Generating props file for code
  console.log('Generating props file...');
  const { fileIds } = await assistantGPT.addFiles(componentPath);

  const assistantCreation = await assistantGPT.init(
    projectName,
    fileIds,
    descriptionPath,
    'props'
  );

  await assistantGPT.response(assistantCreation.id, propsPath);

  await assistantGPT.MarkdownGeneration(
    componentFolder,
    componentName,
    propsPath
  );

  console.log('Documentation generation completed.');
}

main().catch((error) => {
  console.error('Error during the execution of main:', error);
});
