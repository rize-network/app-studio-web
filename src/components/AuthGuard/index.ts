export { AuthGuard, withAuthGuard, RouteAuthGuard, AuthStatus } from './AuthGuard';
