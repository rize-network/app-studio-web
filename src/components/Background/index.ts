export { Background } from './Background';
export type {
  BackgroundProps,
  AuroraBackgroundProps,
  MeteorsProps,
  WallProps,
  ParticlesProps,
  GridProps,
  RipplesProps,
  BackgroundImageProps,
  BackgroundVideoProps,
  BackgroundGradientProps,
  BackgroundStyles,
} from './Background/Background.props';
export type {
  BackgroundEffect,
  AuroraDirection,
  MeteorTiming,
  BorderAnimationStyles,
  StrokeAnimationStyles,
  BackgroundContextType,
  BackgroundSize,
  BackgroundPosition,
  BackgroundRepeat,
  BlendMode,
} from './Background/Background.type';
