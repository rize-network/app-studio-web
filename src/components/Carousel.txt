'use client';
import React, { useEffect, useRef, useState, createContext } from 'react';
import { Horizontal, Vertical } from './Layout';
import { IconButtons } from './Button/examples';
import { Button } from './Button/Button';
import { RightArrowSvg } from 'src/components/Svg/index';

// import { IconLeft, IconRight } from '@app-studio/web/icons';

interface CarouselProps {
  items: JSX.Element[];
  initialScroll?: number;
}

export const CarouselContext = createContext({
  onCardClose: (index: number) => {},
  currentIndex: 0,
});

export const Carousel = ({ items, initialScroll = 0 }: CarouselProps) => {
  const carouselRef = useRef<HTMLDivElement>(null);
  const [canScrollLeft, setCanScrollLeft] = useState(false);
  const [canScrollRight, setCanScrollRight] = useState(true);

  useEffect(() => {
    if (carouselRef.current) {
      carouselRef.current.scrollLeft = initialScroll;
      checkScrollability();
    }
  }, [initialScroll]);

  const checkScrollability = () => {
    if (carouselRef.current) {
      const { scrollLeft, scrollWidth, clientWidth } = carouselRef.current;
      setCanScrollLeft(scrollLeft > 0);
      setCanScrollRight(scrollLeft < scrollWidth - clientWidth);
    }
  };

  const scrollLeft = () =>
    carouselRef.current?.scrollBy({ left: -300, behavior: 'smooth' });
  const scrollRight = () =>
    carouselRef.current?.scrollBy({ left: 300, behavior: 'smooth' });

  return (
    <CarouselContext.Provider
      value={{ onCardClose: () => {}, currentIndex: 0 }}
    >
      <Horizontal
        gap={2}
        alignItems="center"
        width="100%"
        paddingVertical={20}
        position="relative"
      >
        <Button
          onClick={scrollLeft}
          icon={<RightArrowSvg />}
          disabled={!canScrollLeft}
          shape="rounded"
        />
        <Horizontal
          ref={carouselRef}
          onScroll={checkScrollability}
          gap={4}
          overflowX="auto"
          paddingX={10}
        >
          {items.map((item, index) => (
            <Vertical key={index} minWidth="300px">
              {item}
            </Vertical>
          ))}
        </Horizontal>
        <Button
          onClick={scrollRight}
          icon={<RightArrowSvg />}
          disabled={!canScrollRight}
          shape="rounded"
        />
      </Horizontal>
    </CarouselContext.Provider>
  );
};
