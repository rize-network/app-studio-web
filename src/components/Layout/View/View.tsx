import React from 'react';
import type { ViewProps } from 'app-studio';
import { View as $View } from 'app-studio';

export const Top = (props: ViewProps) => (
  <$View marginBottom="auto" {...props} />
);
export const Bottom = (props: ViewProps) => (
  <$View marginTop="auto" {...props} />
);

export const Left = (props: ViewProps) => (
  <$View marginRight="auto" {...props} />
);
export const Right = (props: ViewProps) => (
  <$View marginLeft="auto" {...props} />
);

export const Inline = (props: ViewProps) => (
  <$View display={'inline-flex'} wordBreak="break-word" {...props} />
);

export const View = $View;
