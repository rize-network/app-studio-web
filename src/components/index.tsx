export * from 'src/components/Button/Button';
export * from 'src/components/Form/Checkbox/Checkbox';
export * from 'src/components/Form/CountryPicker/CountryPicker';
export * from 'src/components/Form/DatePicker/DatePicker';
export * from 'src/components/Form/Password/Password';
export * from 'src/components/Form/Select/Select';
export * from 'src/components/Form/Switch/Switch';
export * from 'src/components/Form/Switch/Switch';
export * from 'src/components/Form/TextArea/TextArea';
export * from 'src/components/Form/TextField/TextField';
export * from 'src/components/Layout/Center/Center';
export * from 'src/components/Layout/Horizontal/Horizontal';
export * from 'src/components/Layout/Vertical/Vertical';
export * from 'src/components/Layout/View/View';
export * from 'src/components/Link/Link';
export * from 'src/components/Loader/Loader';
export * from 'src/components/Modal/Modal';
export * from 'src/components/Text/Text';
