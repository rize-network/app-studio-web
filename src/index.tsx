import React from 'react';
import ReactDOM from 'react-dom/client';
import AppProvider from 'src/providers';
import { App } from './App';
import { ComboBoxProvider } from './components/Form/ComboBox/ComboBox/ComboBox.provider';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
root.render(
  <React.StrictMode>
    <AppProvider>
      <ComboBoxProvider>
        <App />
      </ComboBoxProvider>
    </AppProvider>
  </React.StrictMode>
);
