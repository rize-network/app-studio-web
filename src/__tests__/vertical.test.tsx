import React from 'react';
import renderer from 'react-test-renderer';
import { Vertical } from 'src/components';
import { View } from 'src/components/Layout/View/View';

import { cleanup, render, screen } from '@testing-library/react';
afterEach(() => {
  cleanup();
});

test('should render Vertical component without crashing', () => {
  render(
    <Vertical role="Vertical">
      <View width={50} height={50} backgroundColor="theme.primary" />
      <View width={50} height={50} backgroundColor="theme.warning" />
    </Vertical>
  );
});

test('should render Vertical component', () => {
  render(
    <Vertical role="Vertical">
      <View width={50} height={50} backgroundColor="theme.primary" />
      <View width={50} height={50} backgroundColor="theme.warning" />
    </Vertical>
  );
  const VerticalElement = screen.getByRole('Vertical');
  expect(VerticalElement).toBeInTheDocument();
});

test('should render Vertical component with correct class name', () => {
  render(
    <Vertical role="Vertical" className="layout">
      <View width={50} height={50} backgroundColor="theme.primary" />
      <View width={50} height={50} backgroundColor="theme.warning" />
    </Vertical>
  );
  const VerticalElement = screen.getByRole('Vertical');
  expect(VerticalElement).toHaveClass('layout');
});

// test('should render a Vertical layout with display equals to flex', () => {
//   render(
//     <Vertical role="Vertical">
//       <View width={50} height={50} backgroundColor="theme.primary" />
//       <View width={50} height={50} backgroundColor="theme.warning" />
//     </Vertical>
//   );
//   const VerticalElement = screen.getByRole('Vertical');
//   expect(VerticalElement).toHaveStyle({ display: 'flex' });
// });

// test('should render a wrap Vertical layout with flex-wrap equals to wrap', () => {
//   render(
//     <Vertical role="Vertical" flexWrap="wrap">
//       <View width={50} height={50} backgroundColor="theme.primary" />
//       <View width={50} height={50} backgroundColor="theme.warning" />
//     </Vertical>
//   );
//   const VerticalElement = screen.getByRole('Vertical');
//   expect(VerticalElement).toHaveStyle({ flexWrap: 'wrap' });
// });

// test('should render a nowrap Vertical layout with flex-wrap equals to nowrap', () => {
//   render(
//     <Vertical role="Vertical" flexWrap="nowrap">
//       <View width={50} height={50} backgroundColor="theme.primary" />
//       <View width={50} height={50} backgroundColor="theme.warning" />
//     </Vertical>
//   );
//   const VerticalElement = screen.getByRole('Vertical');
//   expect(VerticalElement).toHaveStyle({ flexWrap: 'nowrap' });
// });

// test('should render a wrap reverse Vertical layout with flex-wrap equals to wrap-reverse', () => {
//   render(
//     <Vertical role="Vertical" flexWrap="wrap-reverse">
//       <View width={50} height={50} backgroundColor="theme.primary" />
//       <View width={50} height={50} backgroundColor="theme.warning" />
//     </Vertical>
//   );
//   const VerticalElement = screen.getByRole('Vertical');
//   expect(VerticalElement).toHaveStyle({ flexWrap: 'wrap-reverse' });
// });

// test('should render a justify Vertical layout with justify-content equals to space-between', () => {
//   render(
//     <Vertical role="Vertical" justifyContent="space-between">
//       <View width={50} height={50} backgroundColor="theme.primary" />
//       <View width={50} height={50} backgroundColor="theme.warning" />
//     </Vertical>
//   );
//   const VerticalElement = screen.getByRole('Vertical');
//   expect(VerticalElement).toHaveStyle({ justifyContent: 'space-between' });
// });

// test('should render a reverse Vertical layout with flex-direction equals to row-reverse', () => {
//   render(
//     <Vertical role="Vertical" isReversed>
//       <View width={50} height={50} backgroundColor="theme.primary" />
//       <View width={50} height={50} backgroundColor="theme.warning" />
//     </Vertical>
//   );
//   const VerticalElement = screen.getByRole('Vertical');
//   expect(VerticalElement).toHaveStyle({ flexDirection: 'column-reverse' });
// });

test('Vertical layout to match snapshot', () => {
  const tree = renderer
    .create(
      <Vertical justifyContent="space-between">
        <View width={50} height={50} backgroundColor="theme.primary" />
        <View width={50} height={50} backgroundColor="theme.secondary" />
        <View width={50} height={50} backgroundColor="theme.warning" />
      </Vertical>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
