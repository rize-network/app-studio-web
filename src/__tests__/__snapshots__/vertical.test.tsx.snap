// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical layout to match snapshot 1`] = `
<div
  className="sc-aXZVg bjAhdg"
  display="flex"
>
  <div
    className="sc-aXZVg iKQqOr"
    height={50}
    width={50}
  />
  <div
    className="sc-aXZVg aGRft"
    height={50}
    width={50}
  />
  <div
    className="sc-aXZVg eQfhcF"
    height={50}
    width={50}
  />
</div>
`;
